### PR TITLE
added data event tracker model, interface, and test framework

### PR DESCRIPTION
**Pull request by xavluiz**
This adds the support to track plugin event behavior using the "data_event" entity model.

A lot of the files in this review are just node_module deletions. I've updated the gitignore to ignore the node_modules directory.

I would also like to move the other interfaces into the models directory like I've done for the "EventData".

I've also added a test framework to allow us to create the tracker with a dummy endpoint and provide response feedback.

I'd also like to add a generic typed response to all of the calls to allow the client to always key off of success and failure. This review doesn't have that.

### CODE REVIEW

File: src/models/data_event.model.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// DataEvent interface for the data event payload
```
Comment (bjacobson26) [inline src/models/data_event.model.ts:1]: cool - this is great. good idea to put the other events in model's like this 👍

File: src/models/data_event.model.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  pluginId: number,
+  os: string,
+  version: string,
+  hostname: string,
+  timezone: string
+}
+
+export async function dataEventPayload(params: DataEvent) {
+  return {
+    schema: "iglu:com.software/code_event/jsonschema/1-0-0",
```
Comment (bjacobson26) [inline src/models/data_event.model.ts:18]: i think we should match 1-1 with the snowlow event name. can we name this CodeEvent?

Comment (xavluiz) [reply src/models/data_event.model.ts:18]: yes, we should match it. sorry for the mismatch. I'll update it.

as we discussed in standup I'll change the entity to something related to dom events. DomEvent | BehaviorEvent | StateEvent ?

Comment (xavluiz) [reply src/models/data_event.model.ts:18]: oh and this one isn't related to the "code_event" entity. i think you were confused on that. it's actually called "data_event" in snowplow

Comment (xavluiz) [reply src/models/data_event.model.ts]: updated snowplow entity to dom_event and changed the names and schema reference in this as well

**Pull request merged**